devito/nodes.py

```diff
@@ -394,6 +394,7 @@ def bounds(self, start=None, finish=None):
         available (either statically known or provided through ``start``/
         ``finish``). ``None`` is used as a placeholder in the returned 2-tuple
         if a limit is unknown."""
+        lower = None
```

Comment:
likewise ``upper = None`` ?

Reply:
Tests pass without it. They don't pass without this. Something missing in the tests? :)

Reply:
perhaps...